lib/excon/error.rb

```diff
@@ -48,9 +48,33 @@ def initialize(socket_error = Excon::Error.new)
 
     class InvalidHeaderKey < Error; end
     class InvalidHeaderValue < Error; end
-    class Timeout < Error; end
     class ResponseParse < Error; end
 
+    class Timeout < Error
+      def self.by_type(type, human_name)
+        case type
+        when :connect_read
+          Excon::Errors::ConnectReadTimeout.described_as(human_name)
+        when :connect_write
+          Excon::Errors::ConnectWriteTimeout.described_as(human_name)
+        when :read
+          Excon::Errors::ReadTimeout.described_as(human_name)
+        when :write
+          Excon::Errors::WriteTimeout.described_as(human_name)
+        end
+      end
+
+      def self.described_as(timeout_kind)
+        new("#{timeout_kind.to_s.tr('_', ' ')} timeout reached")
+      end
```

Comment:
I found this to be the most elegant way to give us a little flexibility while maintaining DRY-ness, and keeping exceptions completely vanilla otherwise. (Not overwriting their `initialize`, etc).